src/SAWScript/Builtins.hs

```diff
@@ -949,6 +949,26 @@ provePrintPrim script t = do
     InvalidProof _stats _cex pst -> failProof pst
     UnfinishedProof pst -> failProof pst
 
+provePropPrim ::
+  ProofScript () ->
+  TypedTerm ->
+  TopLevel Theorem
+provePropPrim script t = do
```

Comment:
let's implement a helper that is parametric in a function with type `Term -> IO Prop`, and then let's implement `provePrintPrim`  and `provePropPrim` by using the helper with `predicateToProp sc Universal` and `termToProp sc`